src/acrcssc/azext_acrcssc/helper/_ociartifactoperations.py

```diff
@@ -0,0 +1,233 @@
+import dataclasses
+import tempfile
+import shutil
+import subprocess
+
+from oras.client import OrasClient
+from azure.cli.core.azclierror import AzCLIError
+from azure.cli.command_modules.acr.repository import acr_repository_delete
+from azure.mgmt.core.tools import parse_resource_id
+from jsonschema.exceptions import ValidationError
+from knack.log import get_logger
+from ._constants import (
+    BEARER_TOKEN_USERNAME,
+    CONTINUOUSPATCH_CONFIG_SCHEMA_V1,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1,
+    CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_DRYRUN,
+    CONTINUOUSPATCH_TASK_SCANREGISTRY_NAME,
+    CSSC_WORKFLOW_POLICY_REPOSITORY,
+    SUBSCRIPTION
+)
+from ._utility import (
+    convert_cron_to_schedule,
+    get_task
+)
+
+logger = get_logger(__name__)
+
+
+def create_oci_artifact_continuous_patch(registry, cssc_config_file, dryrun):
+    logger.debug(f"Entering create_oci_artifact_continuouspatching with parameters: {registry.name} {cssc_config_file} {dryrun}")
+
+    oras_client = None
+    temp_artifact_name = None
+    try:
+        oras_client = _oras_client(registry)
+        # we might have to handle the tag lock/unlock for the cssc config file,
+        # to make it harder for the user to change it by mistake
+
+        # the ORAS client can only work with files under the current directory
+        temp_artifact = tempfile.NamedTemporaryFile(
+            prefix="cssc_config_tmp_",
+            mode="w+b",
+            dir=os.getcwd(),
+            delete=False)
+
+        temp_artifact_name = temp_artifact.name
+        user_artifact = open(cssc_config_file, "rb")
+        shutil.copyfileobj(user_artifact, temp_artifact)
+        temp_artifact.close()
+
+        if dryrun:
+            oci_target_name = f"{CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_DRYRUN}"
+        else:
+            oci_target_name = f"{CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1}"
+
+        oras_client.push(
+            target=oci_target_name,
+            files=[temp_artifact.name])
+    except Exception as exception:
+        raise AzCLIError(f"Failed to push OCI artifact to ACR: {exception}")
+    finally:
+        if oras_client:
+            oras_client.logout(hostname=str.lower(registry.login_server))
+        if temp_artifact_name and os.path.exists(temp_artifact_name):
+            os.remove(temp_artifact_name)
+
+
+def get_oci_artifact_continuous_patch(cmd, registry):
+    logger.debug("Entering get_oci_artifact_continuous_patch with parameter: %s", registry.login_server)
+    config = None
+    file_name = None
+    try:
+        oras_client = _oras_client(registry)
+
+        oci_target_name = f"{CSSC_WORKFLOW_POLICY_REPOSITORY}/{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG}:{CONTINUOUSPATCH_OCI_ARTIFACT_CONFIG_TAG_V1}"
+
+        oci_artifacts = oras_client.pull(
```

Comment:
question: is it possible oci_artifacts is none?

Reply:
does not seem like it, since the pull will attempt to get the manifest before the artifact, if it doesn't exist it will throw
https://github.com/oras-project/oras-py/blob/71599b903a05ac09e921af1762dfea9fa6fa8c7c/oras/provider.py#L932C9-L933C43